### DISCUSSION TITLE
Mapping OpenBlas to a custom Matrix Accelerator

### DISCUSSION THREAD

**Discussion by srd8668** [OpenMathLib/OpenBLAS#5015] [Q&A] [answered]
Hi,
i have implemented a Matrix Co processor i want to map Openblas to this coprocessor, can anyone brief about the steps and processes to make that possible currently when i surfed through internet only precompiled Openblas library for existing CPUs are only present.

Regards,
Deepak B

**Comment by martin-frbg** [comment, marked as answer]
Have you checked the (somewhat optimistically named) "Developer manual" in the Wiki (or its equivalent https://github.com/OpenMathLib/OpenBLAS/blob/develop/docs/developers.md in the sources) ? This provides a brief sketch of how to get started when you want to support a new cpu or architecture. (Among other things, there is a set of plain C kernels that should work on any generic architecture you can find a C compiler for, this has been used e.g. to implement the C-Sky cpu "support" we currently have)
You will probably want to replace the GEMM and or GEMV kernels of the underlying architecture with ones that move data to and from your coprocessor. These reside in `kernel/<architecture>` like `kernel/riscv` and the KERNEL file(s) there define which BLAS kernel file to compile for which function. The function call data is passed to them from functions in the interface/ folder through the relevant ones in driver/level2 or driver/level3 that break down big matrix jobs into small blocks suitable for multithreading.

**Reply by srd8668** [reply]
Thanks i will check this out
